SDL/TrackCandidate.h

```diff
@@ -454,7 +472,17 @@ namespace SDL
 
                     unsigned int trackCandidateIdx = alpaka::atomicOp<alpaka::AtomicAdd>(acc, trackCandidatesInGPU.nTrackCandidates,1);
                     alpaka::atomicOp<alpaka::AtomicAdd>(acc, trackCandidatesInGPU.nTrackCandidatesT5,1);
-                    addTrackCandidateToMemory(trackCandidatesInGPU, 4/*track candidate type T5=4*/, quintupletIndex, quintupletIndex, &quintupletsInGPU.logicalLayers[5 * quintupletIndex], &quintupletsInGPU.lowerModuleIndices[5 * quintupletIndex], &quintupletsInGPU.hitIndices[10 * quintupletIndex], -1/*no pixel seed index for T5s*/, quintupletsInGPU.regressionG[quintupletIndex], quintupletsInGPU.regressionF[quintupletIndex], quintupletsInGPU.regressionRadius[quintupletIndex], trackCandidateIdx, quintupletIndex);
+                    trackCandidateIdx -= (nPixTrackCanNotSaved > 0)*nPixTrackCanNotSaved; // Take into account pT5 and pT3 candidates that were not saved
+                    if((trackCandidateIdx >= N_MAX_NONPIXEL_TRACK_CANDIDATES + N_MAX_PIXEL_TRACK_CANDIDATES) || (trackCandidateIdx - nPixTrackCanSaved >= N_MAX_NONPIXEL_TRACK_CANDIDATES))
```

Comment:
The second `if` clause is to make sure that the `N_MAX_NONPIXEL_TRACK_CANDIDATES` limit is satisfied independently. Otherwise, you can have situations where the number of pixel track candidates is below `N_MAX_PIXEL_TRACK_CANDIDATES` so then more than `N_MAX_NONPIXEL_TRACK_CANDIDATES` T5 track candidates can be saved.

Reply:
I would have expected only the "independent satisfaction" of the limit. If the limits are satisfied independent, the additive limit will also be satisfied, and there will be no mixing between which limit counts for which objects.